ludwig/backend/ray.py

```diff
@@ -210,6 +218,36 @@ def shutdown(self):
             ray.kill(handle)
         self.actor_handles.clear()
 
+    class BatchInferModel:
+        def __init__(
+                self,
+                remote_model: RayRemoteModel,
+                predictor_kwargs: Dict[str, Any],
+                output_columns: List[str],
+                features: List[Dict],
+                data_hdf5_fp: str,
+                *args, **kwargs
+        ):
+            self.model = remote_model.load()
```

Comment:
Still not sure about how we're calling `remote_model.load()`, I feel it should cause some issues, but apparently it's not in the tests. My suggestion would be to wrap this in a function that returns a class instead of an instance, so something like this:

```
def get_batch_infer_mode(remote_model, ...):
    class BatchInferModel:
        def __init__(self):
            self.model = remote_mode.load()
            ...
    return BatchInferModel
```

Let me know if this makes sense. Basically, just want to avoid loading the model before we send it to the remote workers.

Reply:
Hey Travis, a couple of questions here:
- If the model is only loaded when sent to remote workers, we won't be able to cache the loaded model in between multiple runs? Following your comment from https://github.com/ludwig-ai/ludwig/pull/1260#discussion_r690647028
- With this implementation of using the function wrapper, does the instantiation of the `BatchInferModel` class happen inside `map_batches`?
- We could also potentially move `remote_model.load()` inside the `__call__` method?

Reply:
Also, I do think it's likely that this was causing issues. E.g., one stacktrace i saw was:

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.6.14/x64/lib/python3.6/site-packages/ray/_private/function_manager.py", line 503, in _load_actor_class_from_gcs
    actor_class = pickle.loads(pickled_class)
  File "/home/runner/work/ludwig/ludwig/ludwig/models/ecd.py", line 294, in __setstate__
    self.set_weights(newstate['weights'])
  File "/opt/hostedtoolcache/Python/3.6.14/x64/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1810, in set_weights
    (self.name, len(weights), expected_num_weights, str(weights)[:50]))
ValueError: You called `set_weights(weights)` on layer "ecd" with a weight list of length 5, but the layer was expecting 3 weights. Provided weights: [array([[ 0.19552892, -0.07009441, -0.17480165, -0...
```

Reply:
Updated!